motoman_driver/MotoPlus/MotionServer.c

```diff
@@ -1563,11 +1567,15 @@ int Ros_MotionServer_GetQueueCnt(Controller* controller, int groupNo)
 BOOL Ros_MotionServer_HasDataInQueue(Controller* controller)
 {
 	int groupNo;
+	int qCnt;
 	
 	for(groupNo=0; groupNo<controller->numGroup; groupNo++)
 	{
-		if(Ros_MotionServer_GetQueueCnt(controller, groupNo) > 0)
+		qCnt = Ros_MotionServer_GetQueueCnt(controller, groupNo);
+		if (qCnt > 0)
 			return TRUE;
+		else if (qCnt == ERROR)

```

Comment:
This could be just an `if`, right (because of the `return`s)? Seeing an `else if` in isolation without an `else` always looks weird to me.

Reply:
Yes, it could.  That change was made by @ted-miller.  Personally, I tend to build the if, else if, else structure based on the conditions that is to be consider (in this case the qCnt value).  It's independently of what occurs after the condition is identified.  So I tend to agree with the way it is.